src/components/Activity/ActivityList.js

```diff
@@ -24,68 +24,96 @@ const ActivityList = ({ apps, activities, keys, clearActivity }) => {
-              <animated.div style={{ ...props, overflow: 'hidden' }}>
-                <ActivityItem
-                  activity={activity}
-                  onClose={() => {
-                    clearActivity(activity.transactionHash)
-                  }}
-                />
-              </animated.div>
-            )
-          }}
-        </Transition>
-      ) : (
-        <div
-          css={`
-            display: flex;
-            flex-direction: column;
-            align-items: center;
-            width: 100%;
-            padding: 64px 32px 24px;
-          `}
-        >
-          <IconEmptyState />
-          <p
-            css={`
-              margin-top: 10px;
-              font-size: 14px;
-              color: ${theme.textSecondary};
-            `}
-          >
-            No activity yet.
-          </p>
-        </div>
-      )}
+  return (
+    <div css="position: relative">
+      <Transition
+        native
+        items={activityItems.length === 0}
+        from={{ opacity: 0 }}
+        enter={{ opacity: 1 }}
+        leave={{ opacity: 0 }}
+        config={springs.smooth}
+      >
+        {show =>
+          show &&
+          (transitionStyles => (
+            <div
+              css={`
+                position: absolute;
+                top: 0;
+                width: 100%;
+                display: flex;
+                flex-direction: column;
+                align-items: center;
+                padding: 64px 32px 24px;
+              `}
+              style={transitionStyles}
+            >
+              <IconEmptyState />
+              <p
+                css={`
+                  margin-top: 10px;
+                  font-size: 14px;
+                  color: ${theme.textSecondary};
+                `}
+              >
+                No activity.
+              </p>
+            </div>
+          ))
+        }
+      </Transition>
+      <Transition
+        native
+        items={activityItems}
+        keys={transitionKeys}
+        trail={100}
+        initial={{
+          opacity: 1,
+          height: 'auto',
+          transform: 'translate3d(0px, 0, 0)',
+        }}
+        from={{
+          opacity: 0,
+          height: 0,
+          transform: 'translate3d(-20px, 0, 0)',
+        }}
+        enter={item => async next => {
+          await next({
+            opacity: 1,
+            height: 'auto',
+            transform: 'translate3d(0px, 0, 0)',
+          })
+          const activityKey = transitionKeys(item)
+          setActivitiesReady({ ...activitiesReady, [activityKey]: true })
+        }}
+        leave={{
+          opacity: 0,
+          height: 0,
+          transform: 'translate3d(-20px, 0, 0)',
+        }}
+        config={springs.smooth}
+      >
+        {activity => transitionStyles => (
+          <animated.div style={{ ...transitionStyles, overflow: 'hidden' }}>
+            <ActivityItem
+              activity={activity}
+              onClose={() => {
```

Comment:
Should this handler be declared out of here?

Reply:
Yes :+1: